### DISCUSSION TITLE
[Question]Search bar doesnt change the colour

### DISCUSSION THREAD

**Discussion by andyargg** [flet-dev/flet#2505] [Q&A] [answered]
### Question

]¿Why my search bar only changes colour when i close or tap in the view instead of change it automatically?
Here is the code
https://github.com/andyargg/APP-MARTIN/tree/main

https://github.com/flet-dev/flet/assets/130791787/9f6eae75-0406-4007-9cb7-6c5a2697720a

### Code sample

```python
class MyApp:
    def __init__(self, page: ft.Page):
        page.window_width=500
      .......................

    def validate_form(self, e):
        for search_boxes in self.search_bars:
            print(search_boxes.obj.bar_bgcolor)
            search_boxes.obj.bar_bgcolor=ft.colors.RED
            print(search_boxes.obj.bar_bgcolor)
        self.page.update()

   def ui(self,e)
     self.next_button = ft.Container(
            padding=10,
            content=ft.Row(
                controls=[
                    
                    ft.ElevatedButton(
                        text="Siguiente",
                        on_click=self.validate_form
                    ),

                ],
                alignment=ft.MainAxisAlignment.SPACE_AROUND
            ),
        )

     --------etc-----------
     self.dropdowns = [self.order_txt, self.cleaning_txt, self.water_txt, self.spare_tire_txt, 
     self.oil_txt, self.car_jack_txt, self.cross_wrench_txt, self.extinguisher_txt, self.padlock_txt]


     self.search_bars = [self.patent_search_bar, self.technician_search_bar]
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Make use of the below properties when creating the `SearchBar` instance:
- https://flet.dev/docs/controls/searchbar#bar_bgcolor
- https://flet.dev/docs/controls/searchbar#view_bgcolor

**Reply by andyargg** [reply]
I fixed it. However when i want to change the text style i have this error: 
TypeError: SearchBar.init() got an unexpected keyword argument 'header_hint_style'.  And this property is in the doc

  def create_search_bar(self):
                anchor = ft.SearchBar(
                    bar_bgcolor='#2c5c74',
                    view_bgcolor='#2c5c74',
                    value="",
                    width=500,
                    view_elevation=4,
                    header_hint_style={
                        'color':'red'
                    },
                    divider_color=ft.colors.AMBER,
                    bar_hint_text=f"Busca {self.data_received}...",
                    view_hint_text=f"Busca {self.data_received}...",
                    bar_leading=ft.IconButton(icon="search", on_click=self.open_anchor),
                    on_change=self.change_item,
                    view_trailing=[
                        ft.IconButton(icon="close",on_click=self.close_anchor),
                    ],

**Reply by ndonkoHenri** [reply, marked as answer]
Oh, sorry, thats an error of mine. The prop is actually called `view_hint_text_style`:
https://github.com/flet-dev/flet/blob/829b0859b5d78917d86813f436bebb2114d92ba4/sdk/python/packages/flet-core/src/flet_core/search_bar.py#L71
Will modify that in the docs.
